waku/protocol/v2/waku_store.nim

```diff
@@ -115,7 +138,13 @@ proc subscription*(proto: WakuStore): MessageNotificationSubscription =
   ## the filter should be used by the component that receives
   ## new messages.
   proc handle(topic: string, msg: WakuMessage) {.async.} =
-    proto.messages.add(msg)
+    try:
+      proto.db.exec(
+        sql"INSERT INTO messages (topic, contentTopic, payload) VALUES (?, ?, ?)",
+        topic, msg.contentTopic, cast[string](msg.payload)
+      )
+    except:
+      warn "dropped message due to sqlite error: " & getCurrentExceptionMsg()
```

Comment:
@kdeme is there a nicer way to do that?

Reply:
I don't know this db code so I don't know what errors can occur here still (and how fatal they are or might be). 
So it would be better to define properly which errors you are catching here, e.g.:

```Nim
except ThisTypeOfError as e:
warn "dropped message due to sqlite error", error = e.msg
```
But I guess you can opt for just `CatchableError` too (I prefer this typically less, as it usually means that nobody properly looked at what can go wrong, but it depends on the case). Definitely don't use only `except` without mentioning a type as this will catch all Exceptions including Defects.